sdk/src/OpenTelemetry/LogRecordProcessor/Simple.hs

```diff
@@ -0,0 +1,66 @@
+{-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE RecordWildCards #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+
+module OpenTelemetry.LogRecordProcessor.Simple (
+  simpleProcessor,
+) where
+
+import Control.Concurrent.Async (async, cancel)
+import Control.Concurrent.Chan.Unagi
+import Control.Exception
+import Control.Monad (forever)
+import qualified Data.HashMap.Strict as H
+import qualified Data.Vector as V
+import OpenTelemetry.Internal.Common.Types
+import OpenTelemetry.Internal.Logs.Types
+
+
+{- | This is an implementation of LogRecordProcessor which passes finished logs and passes the export-friendly ReadableLogRecord
+representation to the configured LogRecordExporter, as soon as they are finished.
+-}
+simpleProcessor :: LogRecordExporter body -> IO (LogRecordProcessor body)
+simpleProcessor LogRecordExporter {..} = do
+  (inChan :: InChan (LogRecord body), outChan :: OutChan (LogRecord body)) <- newChan
+  exportWorker <- async $ forever $ do
+    bracketOnError
+      (readChan outChan)
+      (writeChan inChan)
+      exportSingleLogRecord
```

Comment:
`bracketOnError` should probably only catch asynchronous exceptions in this case, but I think it casts a broader net than that right now. If another exception was caught, it would just keep retrying the export, but [the SDK specifically states](https://opentelemetry.io/docs/specs/otel/logs/sdk/#export) that retrying logic should be handled by the exporter.